apps/comps/rpc/router/agent/get-agent-api-key.ts

```diff
@@ -0,0 +1,55 @@
+import { ORPCError } from "@orpc/server";
+import { z } from "zod";
+
+import { ApiError } from "@recallnet/services/types";
+
+import { base } from "@/rpc/context/base";
+import { userAgentMiddleware } from "@/rpc/middleware/user-agent";
+
+export const getAgentApiKey = base
+  .input(z.object({ agentId: z.string().uuid() }))
+  .use(userAgentMiddleware, (input) => ({ agentId: input.agentId }))
```

Comment:
Ah cool I hadn't done this before. This is how you supply input to the middleware. Makes sense.